src/components/cqlLibraryList/CqlLibraryList.tsx

```diff
@@ -333,107 +427,150 @@ export default function CqlLibraryList({
-        header: "Version",
-        accessorKey: "version",
-        cell: (info) => (
-          <p>
-            {info.row.original.draft && "Draft "}
-            {info.getValue()}
-          </p>
-        ),
+        header: "",
+        cell: (info) => <></>,
+        accessorKey: "",
       },
-      {
-        header: "Actions",
-
-        cell: (info) =>
-          !featureFlags?.LibraryListButtons ? (
-            <Button
-              variant="outline-secondary"
-              style={{ borderColor: "#c8c8c8" }}
-              onClick={(e) => handleOpen(info.row.original, e)}
-              data-testid={`view/edit-cqlLibrary-button-${info.row.original.id}`}
-              aria-label={`CQL Library ${info.row.original.cqlLibraryName} version ${info.row.original.version} draft status ${info.row.original.draft} View / Edit`}
-            >
-              View/Edit
-              <span>
-                <ExpandMoreIcon />
-              </span>
-            </Button>
-          ) : (
-            <Button
-              variant="outline-secondary"
-              style={{ borderColor: "#c8c8c8" }}
-              onClick={() =>
-                navigate(`/cql-libraries/${info.row.original.id}/edit/details`)
-              }
-              data-testid={
-                checkUserCanEdit(
-                  info.row.original.librarySet?.owner,
-                  info.row.original.librarySet?.acls
-                ) && info.row.original.draft
-                  ? `edit-cql-library-button-${info.row.original.id}`
-                  : `view-cql-library-button-${info.row.original.id}`
-              }
-              aria-label={`CQL Library ${info.row.original.cqlLibraryName} version ${info.row.original.version} draft status ${info.row.original.draft} View / Edit`}
-            >
-              {checkUserCanEdit(
-                info.row.original.librarySet?.owner,
-                info.row.original.librarySet?.acls
-              ) && info.row.original.draft
-                ? "Edit"
-                : "View"}
-            </Button>
-          ),
-      }
-    );
-
-    return columnDefs;
-  }, [navigate, featureFlags?.LibraryListCheckboxes]);
+    ];
+  }, [selectedExpandedLibrariesIds, isRowExpanded]);
 
   const table = useReactTable({
     data: cqlLibraryList ?? [],
     columns,
     getCoreRowModel: getCoreRowModel(),
   });
 
-  const selectedLibraries = cqlLibraryList?.filter((library) => {
-    return table
-      .getSelectedRowModel()
-      .rows.find((row) => row.original.id === library.id);
-  });
+  const parentLibraries =
+    cqlLibraryList?.filter((library) => {
+      return table
+        .getSelectedRowModel()
+        .rows.find((row) => row.original.id === library.id);
+    }) || [];
+
+  const expandedLibraries = selectedExpandedLibrariesIds?.map(
+    (expandedLibraryId) => {
+      return expandedSectionData?.find(
+        (data) => data?.id === expandedLibraryId
+      );
+    }
+  );
+
+  const selectedLibraries =
```

Comment:
parentLibraries, expandedLibraries and selectedLibraries all these are not state objects but selectedLibraries is being used in DOM, that means on every render these 3 operations will be executed no matter what. which creates a lot of unnecessary re-renders.